docs/tutorial/src/catalog.py

```diff
@@ -0,0 +1,24 @@
+"""This files represents simple `Application under test`"""
+from typing import List
+
+from attr import Factory, attrib, attrs
+
+
+@attrs
+class Book:
+    author = attrib(type=str)
+    title = attrib(type=str)
+
+
+@attrs
+class Catalog:
+    storage = attrib(default=Factory(list))
+
+    def add_books_to_catalog(self, books: List[Book]):
+        self.storage.extend(books)
+
+    def search_by_author(self, term: str):
+        return filter(lambda book: term in book.author, self.storage)
+
+    def search_by_title(self, term: str):
+        return filter(lambda book: term in book.title, self.storage)
```

Comment:
Please take a look at my catalog class in the second installment of the tutorial series: https://thebddcoach.com/post/completing-the-library-example-and-running-pytest-bdd-ng/

This code is much more elegant than mine, but I wrote mine in a deliberately very simple form so that anyone could understand it. Here, someone must have a much deeper knowledge of Python - they must understand '@attrs', 'attrib', 'filter', 'lambda'... Who is our intended audience? I wrote the tutorial with the clear goal of making it easy to understand; I had no goal to make it sophisticated or clever Python.

I even have comments in mine saying "This isn't the way to write good code, but it is good enough for this tutorial'.